deepchecks/tabular/context.py

```diff
@@ -30,16 +33,118 @@
 ]
 
 
+class _DummyModel:
+    """Contains all the data + properties the user has passed to a check/suite, and validates it seamlessly.
+
+    Parameters
+    ----------
+    train: Dataset
+        Dataset, representing data an estimator was fitted on
+    test: Dataset
+        Dataset, representing data an estimator predicts on
+    y_pred_train: np.ndarray
+        Array of the model prediction over the train dataset.
+    y_pred_test: np.ndarray
+        Array of the model prediction over the test dataset.
+    y_proba_train: np.ndarray
+        Array of the model prediction probabilities over the train dataset.
+    y_proba_test: np.ndarray
+        Array of the model prediction probabilities over the test dataset.
+    """
+
+    train: pd.DataFrame
+    test: pd.DataFrame
+    ind_map: t.Dict[str, t.Dict[t.Any, int]]
+    y_pred: t.Dict[str, np.ndarray]
+    y_proba: t.Dict[str, np.ndarray]
+
+    def __init__(self,
```

Comment:
Think that there should be input validation. Are the inputs ndarrays? are they of the right shape? are the proba values in [0, 1] and sum to 1 on each row? Does the n_cols in proba match the class idx we see in pred? Look in the vision classification data class validations for inspiration.

Reply:
We validate in vision before the user does stuff for our format. Here he just sends ys what ge has

Reply:
But he still can send the wrong thing by mistake, and we should try to warn him if that the case (better than failing with  a cryptic message somewhere inside).
Can perhaps check how sklearn validates this stuff.